src/parsers.rs

```diff
@@ -59,10 +59,22 @@ fn simple_file_output(
         .map_or(
             format!("unknown_{lineno}"),
             |CompileId {
+                 compiled_autograd_id,
                  frame_id,
                  frame_compile_id,
                  attempt,
-             }| { format!("{frame_id}_{frame_compile_id}_{attempt}") },
+             }| {
+                let frame_id_str = frame_id.map_or("-".to_string(), |v| v.to_string());
+                let frame_compile_id_str =
+                    frame_compile_id.map_or("-".to_string(), |v| v.to_string());
+                let attempt_str = attempt.map_or("-".to_string(), |v| v.to_string());
+
+                if let Some(ca_id) = compiled_autograd_id {
+                    format!("{ca_id}_{frame_id_str}_{frame_compile_id_str}_{attempt_str}")
+                } else {
+                    format!("{frame_id_str}_{frame_compile_id_str}_{attempt_str}")
+                }
+            },
```

Comment:
I think this and your earlier site have to be updated in lockstep, so now that it is this complicated you should actually factor this out to a dedicated function.